lib/internal/modules/esm/hooks.js

```diff
@@ -114,6 +116,23 @@ class Hooks {
     this.addCustomLoaders(userLoaders);
   }
 
+  /**
+   * Import and register custom/user-defined module loader hook(s).
+   * @param {string} urlOrSpecifier
+   */
+  async register(urlOrSpecifier) {
+    const moduleLoader = require('internal/process/esm_loader').esmLoader;
+
+    const parentURL = pathToFileURL(getCwdSafe()).href;
+    const keyedExportsSublist = await moduleLoader.import(
+      [urlOrSpecifier],
+      parentURL,
+      kEmptyObject,
+    );
+
+    this.addCustomLoaders(keyedExportsSublist);
+  }
```

Comment:
Can we modify this so it uses this new method? https://github.com/nodejs/node/blob/f14d2e5a0c33996ecbcff1a61a60d787380345c0/lib/internal/modules/esm/utils.js#L153-L162

Reply:
This code does not require using the `privateModuleLoader` instance? Or is it already going to use the same instance when requiring `internal/process/esm_loader`?

```
// Is this going to be the same instance of `privateModuleLoader`?
const moduleLoader = require('internal/process/esm_loader').esmLoader;
```